proxy-attestation-server/src/server.rs

```diff
@@ -195,8 +196,9 @@ async fn nitro_router(nitro_request: web::Path<String>, input_data: String) -> P
     Err(ProxyAttestationServerError::UnimplementedRequestError)
 }
 
-pub fn server<P1, P2>(url: String, ca_cert_path: P1, ca_key_path: P2, debug: bool) -> Result<Server, String>
+pub fn server<U, P1, P2>(url: U, ca_cert_path: P1, ca_key_path: P2, debug: bool) -> Result<Server, String>
```

Comment:
Is there something that requires `U`, `P1`, and `P1` instead of using the types directly? Two levels of indirection to find out what types the parameters are seems a bit much if it can be avoided.

Reply:
Strictly no, these could be changed to a bunch of `&str`s. But that would move more work into the user to convert down to `&str` when HttpServer::bind/File::open are going to take U:ToSocketAddrs/P:AsRef<Path> anyways. It would also be a little bit less type safe, using `&str` for everything instead of specific traits. 

As for the indirection, that's sort of on Rust's syntax. The traits could be inlined in this case (not always), but the syntax of traits require at least one indirection without changing the behavior of the API:

``` rust
pub fn server<
    U: ToSocketAddrs,
    P1: AsRef<path::Path>,
    P2: AsRef<path::Path>
>(
    url: U,
    ca_cert_path: P1,
    ca_key_path: P2,
    debug: bool
) -> Result<Server, String> {
    blablabla
```

Reply:
The functionality LGTM. Cannot check the code.